library/std/src/os/unix/net/udp.rs

```diff
@@ -0,0 +1,249 @@
+use super::IpAncillary;
+use crate::{
+    io::{self, IoSlice, IoSliceMut},
+    net::{SocketAddr, ToSocketAddrs, UdpSocket},
+    sys_common::AsInner,
+};
+
+impl UdpSocket {
```

Comment:
Unless I missed a policy change simply adding platform-specific methods to a cross-platform struct isn't allowed and sealed extension traits should be used instead.